### PR TITLE
[Distributed] [16/N] Fix clang-tidy warnings in torch/csrc/distributed/c10d

### PR DESCRIPTION
**Pull request by cyyever**
Follows #137072

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

File: test/cpp/c10d/HashStoreTest.cpp [hunk 1]
```diff
@@ -60,16 +60,16 @@ void stressTestStore(std::string prefix = "") {
   std::vector<std::thread> threads;
   c10d::test::Semaphore sem1, sem2;
   auto hashStore = c10::make_intrusive<c10d::HashStore>();
-  c10d::PrefixStore store(prefix, hashStore);
+  c10d::PrefixStore store(std::move(prefix), hashStore);
 
   for (C10_UNUSED const auto i : c10::irange(numThreads)) {
-    threads.emplace_back(std::thread([&] {
```
Comment (Skylion007) [inline test/cpp/c10d/HashStoreTest.cpp:66]: Is this really any cleaner / more efficient without std::thread?

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp [hunk 2]
```diff
@@ -2318,7 +2315,7 @@ std::shared_ptr<NCCLComm> ProcessGroupNCCL::getNCCLComm(
     C10D_NCCL_CHECK(ncclGroupStart(), std::nullopt);
   }
 
-  ncclStreams_.emplace(deviceKey, std::move(streamVal));
+  ncclStreams_.emplace(deviceKey, streamVal);
```
Comment (Skylion007) [inline torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:2318]: Streamval was trivially copyable?

**Review by Skylion007 — APPROVED** [review]

File: torch/csrc/distributed/c10d/NCCLUtils.hpp [hunk 3]
```diff
@@ -626,9 +626,9 @@ struct NCCLTraceBuffer {
     std::optional<c10::time_t> time_discovered_completed_;
 
     // size information for input/output tensors
-    c10::SmallVector<int, 4> input_dims_;
+    c10::SmallVector<int64_t, 4> input_dims_;
     std::vector<c10::ScalarType> input_dtypes_;
-    c10::SmallVector<int, 4> output_dims_;
+    c10::SmallVector<int64_t, 4> output_dims_;
```
[the remark this answers is not in the document]
Comment (cyyever) [reply torch/csrc/distributed/c10d/NCCLUtils.hpp:631]: Dims in Pytorch are usually int64_t.

File: test/cpp/c10d/HashStoreTest.cpp (hunk 1 above)
Comment (cyyever) [reply test/cpp/c10d/HashStoreTest.cpp:66]: It's a concurrency test

File: torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp (hunk 2 above)
Comment (cyyever) [reply torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:2318]: Yes.

**Pull request closed without merging**